packages/mobx-react-lite/__tests__/observer.test.tsx

```diff
@@ -18,6 +23,7 @@ function runTestSuite(mode: "observer" | "useObserver") {
             return observer(component)
         } else {
             const c = (props: P) => {
+                consoleWarnMock = jest.spyOn(console, "warn").mockImplementation(() => {})
                 return useObserver(() => {
```

Comment:
@urugator I had to just silence console warnings here because the function runTestSuite is called twice with different arguments and the snapshots didn't work for me.

https://github.com/mobxjs/mobx/blob/main/packages/mobx-react-lite/__tests__/observer.test.tsx#L481-L482

Since it is deprecated API I think it's better to leave it as is and remove `useObserver` in next major verions